### PR TITLE
New ROSbot PROTO with associated devices PROTOs

### PR DESCRIPTION
**Pull request by ygoumaz**
Implementation of the `ROSbot` PROTO in Webots.

The robot embeds the following sensors which are also created as PROTO files:
- `Astra` RGBD Camera.
- `RpLidarA2` Lidar.
- `Mpu-9250` Inertial Measurement Unit (IMU).

### CODE REVIEW

**Comment by omichel** [comment]
As a quick first review, it seems that some meshes are too much faceted. We should smooth them. I believe this can be done in Blender on the OBJ files.

**Comment by ygoumaz** [comment]
obj files now all have a smooth shading applied to them.

**Comment by omichel** [comment]
> .obj files now all have a smooth shading applied to them.

Thank you, that's nice. However, I still have a few comments on the 3D models, like the wheels now look weird, the connectors should have a different color from the body, etc. However, I believe it is simpler that I show them to you one-by-one.

**Comment by ygoumaz** [comment]
The PR is again ready for review. The ROSbot PROTO has been significantly improved visually. Any feedback to improve the documentation and the sample world is welcome.

**Comment by omichel** [comment]
It looks very good to me, thank you. I just noticed strange triangle patterns on the red body of the robot which was not present in the previous version I believe:
rosbot
Could this be caused by the modified UV mapping?

**Comment by ygoumaz** [comment]
I fixed the reflection problem on the top red part of the robot. I also applied the same solution to the wheel rims and it improved a lot the mapping of the aluminium visually.

The solution is to select all the corresponding faces of the Mesh in Blender and to "UV unwrap faces". It allows to apply the mapping to all faces rather than to each triangle separately.

**Comment by omichel** [comment]
It looks very nice now. Thank you.
There are still pending comments on this PR to be addressed.

**Review by omichel — CHANGES_REQUESTED** [review]

File: projects/devices/orbbec/protos/Astra.proto [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#VRML_SIM R2023a utf8
+# license: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+# license url: https://cyberbotics.com/webots_assets_license
+# documentation url: https://www.cyberbotics.com/doc/guide/camera-sensors#astra
+# Model of the Astra RGBD Camera developed by Orbbec.
```
Comment (omichel) [inline projects/devices/orbbec/protos/Astra.proto:5]:
```suggestion
# Model of the Astra RGBD camera developed by Orbbec.
```

File: projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  /* rotational lidar */
+  WbDeviceTag lidar;
+  /* IMU */
+  WbDeviceTag accelerometer, gyro, compass;
+  /* distance sensors */
+  WbDeviceTag distance_sensors[4];
+  double distance_sensors_value[4];
+
+  // set empirical coefficients for collision avoidance
+  double coefficients[2][2] = {{15.0, -9.0}, {-15.0, 9.0}};
```
Comment (omichel) [inline projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c:54]:
```suggestion
  const double coefficients[2][2] = {{15.0, -9.0}, {-15.0, 9.0}};
```

File: projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+int main(int argc, char *argv[]) {
+  /* define variables */
+  /* motors */
+  WbDeviceTag front_left_motor, front_right_motor, rear_left_motor, rear_right_motor, front_left_position_sensor,
+    front_right_position_sensor, rear_left_position_sensor, rear_right_position_sensor;
+  double avoidance_speed[2];
+  double base_speed = 6.0;
+  double motor_speed[2];
+  /* camera RGBD */
```
Comment (omichel) [inline projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c:43]:
```suggestion
  /* RGBD camera */
```

File: projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+#define TIME_STEP 32
+#define MAX_VELOCITY 26
+
+int main(int argc, char *argv[]) {
+  /* define variables */
+  /* motors */
+  WbDeviceTag front_left_motor, front_right_motor, rear_left_motor, rear_right_motor, front_left_position_sensor,
+    front_right_position_sensor, rear_left_position_sensor, rear_right_position_sensor;
+  double avoidance_speed[2];
+  double base_speed = 6.0;
```
Comment (omichel) [inline projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c:41]:
```suggestion
  const double base_speed = 6.0;
```

File: projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  // set empirical coefficients for collision avoidance
+  double coefficients[2][2] = {{15.0, -9.0}, {-15.0, 9.0}};
+
+  int i, j;
+
+  /* initialize Webots */
+  wb_robot_init();
+
+  /* get a handler to the motors and set target position to infinity (speed
+   * control). */
```
Comment (omichel) [inline projects/robots/husarion/rosbot/controllers/rosbot/rosbot.c:62]:
```suggestion
  /* get a handler to the motors and set target position to infinity (speed control). */
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — DISMISSED** [review]
Thank you. Everything looks good to me now.

**Comment by ygoumaz** [comment]
Some `WorldInfo` fields of the sample world were not correct.

**Review by omichel — APPROVED** [review]
:+1:

**Pull request merged**